src/CBLCollection_Internal.hh

```diff
@@ -134,8 +134,15 @@ public:
     
     void createArrayIndex(slice name, CBLArrayIndexConfiguration config) {
         C4IndexOptions options = {};
-        // TODO Set options for array path when LiteCore API is ready
-        _c4col.useLocked()->createIndex(name, config.expressions,
+        options.unnestPath = (char*)config.path.buf;
+        
+        auto exprs = config.expressions;
+        if (!exprs.buf && config.expressionLanguage == kCBLJSONLanguage) {
+            exprs = FLStr("[]");
+        }
+        
+        config.expressions = exprs;
+        _c4col.useLocked()->createIndex(name, exprs,
```

Comment:
similar observations here. 

For .NET, Jim is calling createIndex using `C4QueryLanguage.N1QLQuery`, not by passing expressionLanguage.

For iOS, I am doing same as you.

Reply:
Not sure what it means but iOS only support N1QLQuery language for the IndexConfiguration (Not IndexBuilder) API.